Source/drlg_l3.cpp

```diff
@@ -792,11 +793,11 @@ static void DRLG_L3Pool()
 				found = TRUE;
 			}
 			poolchance = random_(0, 100);
-			for (j = duny - totarea; j < duny + totarea; j++) {
-				for (i = dunx - totarea; i < dunx + totarea; i++) {
+			for (j = std::max(0, std::min(DMAXY - 1, duny - totarea)); j < duny + totarea; j++) {

```

Comment:
This should be more like:

```c++
for (j = std::max(duny - totarea, 0); j < std::min(duny + totarea, DMAXY); ++j) {
```

Similar in other places

Reply:
Ok how about now :)